Drone/drone_base.py

```diff
@@ -0,0 +1,361 @@
+
+    cnt = 0
+
+    def __init__(self):
+        self.vehicle = None
+        self.devices = []
+        self.connected = False
+        self.logger = logging.getLogger(__name__)
+
+    def connect(self, isInSimulator):
+        """
+        Attempts to connect to Ardupilot
+
+        Parameters
+        ----------
+        isInSimulator: Boolean
+            True if the program is intended to run in the Gazebo simulator
+            and False if the program is being run on the real-life drone
+
+        Precondition:
+        ----------
+        None
+
+        Postcondition:
+        ----------
+        Upon a successful connection, self.connected is set to True.
+        The drone is now suitable to be armed.
+
+        Returns:
+        ----------
+        None
+        """
+        if isInSimulator:
+            self.vehicle = dronekit.connect(c.CONNECTION_STRING_SIMULATOR, wait_ready=True, 
+                status_printer=None)
+            self.logger.info(threading.current_thread().name + ": Connecting to the simulated ardupilot")
+        else:
+            self.vehicle = dronekit.connect(c.CONNECTION_STRING_REAL, wait_ready=True, 
+                status_printer=None)
+            self.logger.info(threading.current_thread().name + ": Connecting to the real-life ardupilot")
+
+        if self.vehicle is not None:
+            self.connected = True
+
+    def altitude(self):
+        """
+        Simple wrapper function to retrieve the distance being read by 
+        the drone's rangefinder
+
+        Returns:
+        ----------
+        double
+            The distance from the ground
+        """
+        return self.vehicle.rangefinder.distance
+
+    def altitude_failsafe(self):
+        """
+        Simple wrapper function to retrieve the distance being read by 
+        the drone's barometer
+
+        Returns:
+        ----------
+        double
+            The distance from the ground
+        """
+        return self.vehicle.location.global_relative_frame.alt
+
+    def arm(self, timeout = 60):
+        """
+        Attempts to arm the drone for flight
+
+        Parameters
+        ----------
+        timeout: Integer (optional)
+            The duration in seconds that arming should be attempted before
+            giving up and returning
+
+        Precondition:
+        ----------
+        self.connect has been called and succeeded.
+
+        Postcondition:
+        ----------
+        Upon successful arm, self.vehicle.armed is True, and false otherwise.
+        The drone is now suitable to take off.
+
+        Returns:
+        ----------
+        None
+        """
+        start = time.time()
+        self.vehicle.mode = VehicleMode(c.GUIDED_MODE)
+
+        self.logger.info(threading.current_thread().name + ": Arming")
+        start = time.time()
+        while (not self.vehicle.armed) and (time.time() - start < timeout):
+            self.vehicle.armed = True
+            time.sleep(1)
```

Comment:
Consider
```suggestion
            time.sleep(c.ARM_RETRY_DELAY)
```

Reply:
I have also added the constant c.MAV_MESSAGE_RESEND_DELAY to fix other similar issues.